### ISSUE TITLE
Getting Error [Field 'issues' should not be null.] for type as suite.

### ISSUE THREAD

**Issue by saurabhsrivastava2009** [reportportal/reportportal#2009]
Hi Team,

I was following the documentation present at https://reportportal.io/docs/dev-guides/ReportingDevelopersGuide/ to create a dashboard. I'm able to create the launch, but while adding the suite it is throwing the following error
`
{"errorCode":4001,"message":"Incorrect Request. [Field 'issues' should not be null.] "}`

Whereas as per the document, there is no field called '**issues**' for _type as suite_. A similar error is there for type as test, there it fails with
`{"errorCode":4001,"message":"Incorrect Request. [Field 'endTime' should not be null.] "}`

and I don't see this field/attribute present in the table ( docs).

Could you help with the resolution of this issue?

Version 5.7.4

Update :
Tried to add issue key to the body for debugging, but still it did not worked. Here is the error
`{"errorCode":4001,"message":"Incorrect Request. JSON parse error: Cannot deserialize value of type `java.util.ArrayList<com.epam.ta.reportportal.ws.model.issue.IssueDefinition>` from String value (token `JsonToken.VALUE_STRING`); nested exception is com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.ArrayList<com.epam.ta.reportportal.ws.model.issue.IssueDefinition>` from String value (token `JsonToken.VALUE_STRING`)\n at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream); line: 1, column: 48] (through reference chain: com.epam.ta.reportportal.ws.model.issue.DefineIssueRQ[\"issues\"])"}`

**Comment by MaryiaRymarchyk** [comment]
Hi @saurabhsrivastava2009, could you please clarify some moments here:
- are you talking about **_dashboard_** creation  or  start root(suite) **_item_**?
- which agent do you use for it (postman etc.)?
- could you share the Body of the request which leads to such response with error.

**Comment by saurabhsrivastava2009** [comment]
Hi @MaryiaRymarchyk 
PFB answers to your questions
- I'm talking about starting the root(suite) item.
- I'm using curl command
-  `'{"name":"SomethingExtract","description":"This is the test class","startTime":"1684489101.1982317","type":"suite","launchUuid":"6d29b41c-c22f-421e-b66c-e3fa6e8575fa"}'`

**Comment by MaryiaRymarchyk** [comment]
@saurabhsrivastava2009, based on your answers, Request body for Suite creation looks correct. 
There is no any mandatory 'issues' field for type 'suite'. It seems that Request body for suite is not the cause of the issue.
Regarding  'deserialize' error - it is know issue and it will be fixed in next versions, however for now you can try workaround solutions from the ticket - #1970. Probably if you fix 'deserialize' error you will be able to debug the problem with 'suite'.
Could you please try one more time after fixing 'deserialize'error and get back to us with updates?

**Comment by saurabhsrivastava2009** [comment]
Hi @MaryiaRymarchyk , the issue seems to be related to POST, and now it is working. But after that I'm getting error while posting the log
```
curl -k -X POST https://something/api/v1/Automation_Dashboard/log -H 'Content-Type: application/json' -H 'Authorization: Bearer something145ssd44' -d '{"launchUuid":"8ce405e1-fbf3-40cc-947a-9e45e44d5f49","itemUuid":"07d967db-c840-416b-a67c-dad63139c51c","message":"Expected Class.Method(param1, param2, false) to be True, but found False.
2023-05-23 12:24:56.473 |
2023-05-23 12:24:56.473 |   Stack Trace:    at FluentAssertions.Execution.LateBoundTestFramework.Throw(String message)
2023-05-23 12:24:56.473 |      at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
2023-05-23 12:24:56.473 |      at FluentAssertions.Execution.CollectingAssertionStrategy.ThrowIfAny(IDictionary`2 context)
2023-05-23 12:24:56.473 |      at somepackage.Test.testfolder.testclass.LevelExtract() in /src/somepackage.Test/testfolder/testclass.cs:line 152
2023-05-23 12:24:56.473 |
2023-05-23 12:24:56.473 |   https://screenshots.services.af.k8s.blackline.corp/screenshots_somescreenshot_29376b6b27704892aaba68186d7efa11/ss_22May20232340.png","level":"error"}'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1634    0   559  100  1075    383    737  0:00:01  0:00:01 --:--:--  1121{"errorCode":4001,"message":"Incorrect Request. JSON parse error: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value\n at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream); line: 1, column: 265] (through reference chain: com.epam.ta.reportportal.ws.model.log.SaveLogRQ[\"message\"])"}
```

**Comment by MaryiaRymarchyk** [comment]
Hi @saurabhsrivastava2009, This error can happen if you have a newline (or other control character) in a JSON string literal.
You can validate your JSON via https://jsonlint.com/ 
The solution is to escape control characters  (  /n has to be changed to //n )
Please try and let us know about results.

**Comment by saurabhsrivastava2009** [comment]
Hi @MaryiaRymarchyk 
I did change it
```
def exceptionWithSlashN= "something\n something else"
def simpleExceptionMessage = exceptionWithSlashN.replaceAll("\\n", "\\\n")
```
But still it throws the same error

**Comment by MaryiaRymarchyk** [comment]
Hi @saurabhsrivastava2009,
can you try with  '/' slash , not '\' slash . Or try to remove all new lines from the stack trace - just to make sure that the issue in that.

**Comment by MaryiaRymarchyk** [comment]
Hi @saurabhsrivastava2009 do you have any updates here?

**Comment by saurabhsrivastava2009** [comment]
Hi @MaryiaRymarchyk 
I tried but it did not work..!
I also tried with _System.lineSeparator()_, but this also did not work

**Comment by MaryiaRymarchyk** [comment]
@saurabhsrivastava2009 , so you still get the same error on posting log ?
"errorCode":4001,"message":"Incorrect Request. JSON parse error: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value\n at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream); line: 1, column: 265] (through reference chain: com.epam.ta.reportportal.ws.model.log.SaveLogRQ[\"message\"])

It is really strange as it clearly related to illegal character which should be escaped. 
Can you please share your current request Body  and latest error.

**Comment by saurabhsrivastava2009** [comment]
No, I'm not getting this error, instead it is printing "n" in the message. So I changed it `System.lineSeparator()` but this is throwing the error  `Scripts not permitted to use staticMethod java.lang.System lineSeparator. Exception while posting to report portal:org.****ci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use staticMethod java.lang.System lineSeparator`

**Comment by saurabhsrivastava2009** [comment]
The exception Message is something like this

Here the message that we are getting from the DB is 
`at OpenQA.Selenium.Support.UI.DefaultWait1.ThrowTimeoutException(String exceptionMessage, Exception lastException)\n   at OpenQA.Selenium.Support.UI.DefaultWait1.Until[TResult](Func2 condition, CancellationToken token)\n   at OpenQA.Selenium.Support.UI.DefaultWait1.Until[TResult](Func2 condition\)n   at `

So I removed the \n with \\n and this gave me the below string, which doesn't throw any error, but is not formatted properly as well.
`at OpenQA.Selenium.Support.UI.DefaultWait1.ThrowTimeoutException(String exceptionMessage, Exception lastException)n   at OpenQA.Selenium.Support.UI.DefaultWait1.Until[TResult](Func2 condition, CancellationToken token)n   at OpenQA.Selenium.Support.UI.DefaultWait1.Until[TResult](Func2 condition)n   at `

I tried with '/' as well, but it did not work. So I went with System.lineSeparator() and it throws the error : Scripts not permitted to use staticMethod java.lang.System lineSeparator. Exception while posting to report portal:org.****ci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use staticMethod java.lang.System lineSeparator
